packages/router/src/nuxt/index.js

```diff
@@ -27,107 +28,122 @@
-      wildcard: true,
-      ...((this.options || {}).druxt || {}).router,
-      ...moduleOptions,
+const DruxtRouterNuxtModule = defineNuxtModule({
+  meta: {
+    name: 'druxt-router',
+  },
+  defaults: {
+    baseUrl: '',
+    endpoint: '/jsonapi',
+  },
+
+  async setup(moduleOptions, nuxt) {
+    // Set default options.
+    const options = {
+      baseUrl: moduleOptions.baseUrl,
+      ...nuxt.options?.druxt || {},
+      router: {
+        pages: (await existsSync(resolve(nuxt.options.srcDir, nuxt.options.dir.pages))),
+        wildcard: true,
+        ...nuxt.options?.druxt?.router,
+        ...moduleOptions,
+      }
     }
-  }
 
-  // Add dependant modules.
-  await this.addModule(['druxt', options])
+    // Add dependant modules.
+    await installModule('druxt/nuxt', options, nuxt)
 
-  // Register components directories.
-  this.nuxt.hook('components:dirs', dirs => {
-    dirs.push({ path: join(__dirname, '../dist/components') })
-  })
+    // Register components directories.
+    nuxt.hook('components:dirs', dirs => {
+      dirs.push({ path: join(__dirname, '../dist/components') })
+    })
 
-  // Add Druxt router custom wildcard route.
-  if (options.router.wildcard) {
-    // Ignore page routes.
-    if (!options.router.pages) {
-      this.nuxt.hook('build:before', () =>
-        this.nuxt.options.build.createRoutes = () => []
-      )
-    }
+    // Add Druxt router custom wildcard route.
+    if (options.router.wildcard) {
+      // Ignore page routes.
+      if (!options.router.pages) {
+        nuxt.hook('build:before', () =>
+          nuxt.options.build.createRoutes = () => []
+        )
+      }
 
-    // Add route template.
-    this.addTemplate({
-      src: resolve(__dirname, '../templates/component.js'),
-      fileName: 'components/druxt-router.js',
-      options
-    })
+      // Add route template.
+      addTemplate({
+        src: resolve(__dirname, '../templates/component.js'),
+        fileName: 'components/druxt-router.js',
+        options,
+        write: true
+      })
 
-    // Fetch languages.
-    let languages = []
-    const druxt = new DruxtClient(options.baseUrl, {
-      ...options,
-      // Disable API Proxy, as Proxies aren't available at build.
-      proxy: { ...options.proxy || {}, api: false },
-    })
-    const languageResourceType = 'configurable_language--configurable_language'
-    if (((await druxt.getIndex(languageResourceType)) || {}).href) {
-      const query = new DrupalJsonApiParams().addFields(languageResourceType, ['drupal_internal__id'])
-      languages = (await druxt.getCollectionAll(languageResourceType, query) || [])
-        .map((o) => o.data)
-        .flat()
-        .filter((o) => !['und', 'zxx'].includes(((o || {}).attributes || {}).drupal_internal__id))
-    }
+      // Fetch languages.
+      let languages = []
+      const druxt = new DruxtClient(options.baseUrl, {
+        ...options,
```

Comment:
This code is not new, but has been moved from a different file.

However, it would make sense to expand on the tests, so please open a new issue with some details.